objectRecognitionSource/src/main/java/edu/ml/tensorflow/util/ImageUtil.java

```diff
@@ -89,7 +116,33 @@ public void saveImage(final BufferedImage image, final String target) {
                 try {
                     File imgFile = File.createTempFile("tmp", ".jpg");
                     imgFile.deleteOnExit();
-                    ImageIO.write(image, "jpg", imgFile);
+                    if (longEdge == 0) {
+                        ImageIO.write(image, "jpg", imgFile);
+                    } else {
+                        int width = image.getWidth();
+                        int height = image.getHeight();
+                        if (longEdge > Math.max(width, height)) {
+                            LOGGER.error("The longEdge value is too large, resizing cannot enlarge the original image. "
+                                    + "Original image resolution will not be changed");
+                            ImageIO.write(image, "jpg", imgFile);
+                            longEdge = 0;
+                        } else {
+                            if (width > height) {
+                                double ratio = (double) longEdge/width;
+                                int newHeight = (int) (height * ratio);
+                                BufferedImage resizedImage = resizeImage(longEdge, newHeight, image);
+                                ImageIO.write(resizedImage, "jpg", imgFile);
+                            } else if (height > width) {
+                                double ratio = (double) longEdge/height;
+                                int newWidth = (int) (width * ratio);
+                                BufferedImage resizedImage = resizeImage(newWidth, longEdge, image);
+                                ImageIO.write(resizedImage, "jpg", imgFile);
+                            } else {
+                                BufferedImage resizedImage = resizeImage(longEdge, longEdge, image);
+                                ImageIO.write(resizedImage, "jpg", imgFile);
+                            }
+                        }
+                    }
```

Comment:
Seems like we should be able to refactor all this stuff in a method to call.  I think you're working on this.

Reply:
Yep, I'm working on this.